zerver/migrations/0465_rename_followup_day_email_templates.py

```diff
@@ -0,0 +1,59 @@
+# Generated by Django 4.2.2 on 2023-07-11 10:45
+
+import orjson
+from django.db import migrations
+from django.db.backends.base.schema import BaseDatabaseSchemaEditor
+from django.db.migrations.state import StateApps
+
+WELCOME = 1
+
+
+def update_for_followup_day_email_templates_rename(
+    apps: StateApps, schema_editor: BaseDatabaseSchemaEditor
+) -> None:
+    ScheduledEmail = apps.get_model("zerver", "ScheduledEmail")
+    scheduled_onboarding_emails = ScheduledEmail.objects.filter(type=WELCOME)
+
+    for email in scheduled_onboarding_emails:
+        data = orjson.loads(email.data)
```

Comment:
The email data with the `"template_prefix"` information is in a text field from the parent class, `AbstractScheduledJob`. As there wasn't a way to query the database for that specific data, I just wrote a Python loop over the potential emails that need to be updated.

Reply:
I think that's a fine option for this case -- but it _is_ possible to do in the database, despite (currently) being a `text` and not a `jsonb` field:

```python3
from django.db.models import F, Func, JSONField, TextField, Value
from django.db.models.functions import Cast

from zerver.models import ScheduledEmail

sched = ScheduledEmail.objects.annotate(as_jsonb=Cast("data", JSONField())).filter(
    type=WELCOME,
    as_jsonb__template_prefix="zerver/emails/followup_day2"
)
sched.update(
    data=Cast(
        Func(
            F("as_jsonb"),
            Value(["template_prefix"]),
            Value("onboarding_zulip_topics", JSONField()),
            function="jsonb_set",
        ),
        TextField(),
    )
)
```

Only very lightly tested, so take with appropriate grain of salt.

Reply:
Okay. Rewrote the migration based on the above and wrote a basic migration test. And did some manual testing in the db_shell of the dev environment.